File: VB/GridWithFilterPanelButtons/CustomGridControl/CustomGridControlViewInfoRegistrator.vb
Imports DevExpress.XtraGrid
Imports DevExpress.XtraGrid.Registrator
Imports DevExpress.XtraGrid.Views.Base
Imports DevExpress.XtraGrid.Views.Base.ViewInfo
Imports System

Namespace GridWithFilterPanelButtons

    Public Class CustomGridControlViewInfoRegistrator
        Inherits GridInfoRegistrator

        Public Overrides Function CreatePainter(ByVal view As BaseView) As BaseViewPainter
            Return New CustomGridControlViewPainter(TryCast(view, CustomGridControlView))
        End Function

        Public Overrides Function CreateView(ByVal grid As GridControl) As BaseView
            Return New CustomGridControlView(grid)
        End Function

        Public Overrides Function CreateViewInfo(ByVal view As BaseView) As BaseViewInfo
            Return New CustomGridControlViewInfo(TryCast(view, CustomGridControlView))
        End Function

        Public Overrides ReadOnly Property ViewName() As String
            Get
                Return "CustomGridControlView"
            End Get
        End Property
    End Class
End Namespace

File: CS/GridWithFilterPanelButtons/CustomGridControl/CustomGridControlViewInfoRegistrator.cs
using DevExpress.XtraGrid;
using DevExpress.XtraGrid.Registrator;
using DevExpress.XtraGrid.Views.Base;
using DevExpress.XtraGrid.Views.Base.ViewInfo;
using System;

namespace GridWithFilterPanelButtons
{

    public class CustomGridControlViewInfoRegistrator : GridInfoRegistrator
    {
        public override BaseViewPainter CreatePainter(BaseView view)
        {
            return new CustomGridControlViewPainter(view as CustomGridControlView);
        }

        public override BaseView CreateView(GridControl grid)
        {
            return new CustomGridControlView(grid);
        }

        public override BaseViewInfo CreateViewInfo(BaseView view)
        {
            return new CustomGridControlViewInfo(view as CustomGridControlView);
        }

        public override string ViewName => "CustomGridControlView";
    }
}